src/amo/components/Home/index.js

```diff
@@ -46,13 +52,19 @@ export class HomeBase extends React.Component {
   static propTypes = {
     clientApp: PropTypes.string.isRequired,
     dispatch: PropTypes.func.isRequired,
+    errorHandler: PropTypes.object.isRequired,
     i18n: PropTypes.object.isRequired,
+    popularExtensions: PropTypes.array.isRequired,
   }
 
   componentWillMount() {
-    const { dispatch } = this.props;
+    const { dispatch, errorHandler, popularExtensions } = this.props;
 
     dispatch(setViewContext(VIEW_CONTEXT_HOME));
+
+    if (popularExtensions.length === 0) {
```

Comment:
We should get out of the habit of using empty arrays to indicate that data has not yet loaded. It is possible to receive an empty array as a legitimate API result and in that case your code will get stuck in an infinite dispatch loop. It's unlikely for the popular extensions API to return an empty array but when using a local development server that isn't configured right, it could happen. It would be safer to do this:

```js
if (popularExtensions === null) {
  dispatch(fetchHomeAddons(...));
}
```

And then you'd need to initialize the reducer like that too:

```js
export const initialState: HomeState = {
  popularExtensions: null,
};
```

Reply:
That's a good point, and we probably do this elsewhere. I've filed #3372.

Reply:
Indeed. For my defense, I did it on purpose because there are always popular extensions (in theory) and I removed it in the upcoming PR about the 2nd shelf in favor of a `resultsLoaded` flag.

Reply:
That being said, I should not have done that.